fuzz/punycode.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2022 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include "crypto/punycode.h"
+#include "internal/nelem.h"
+#include <openssl/crypto.h>
+#include "fuzzer.h"
+
+#include <stdio.h>
+#include <string.h>
+
+int FuzzerInitialize(int *argc, char ***argv)
+{
+    return 1;
+}
+
+int FuzzerTestOneInput(const uint8_t *buf, size_t len)
+{
+    char *b;
+    unsigned int out[16], outlen = OSSL_NELEM(out);
+    char outc[16];
+
+    b = OPENSSL_malloc(len + 1);
+    if (b != NULL) {
+        memcpy(b, buf, len);
+        b[len] = '\0';
+        ossl_punycode_decode(b, len, out, &outlen);
```

Comment:
Why doesn't this just pass buf? I see no reason to alloc b with an extra \0.

Reply:
It seems to be needed for ossl_a2ulabel, but not for ossl_punycode_decode

Reply:
The fuzzer doesn't \0 terminate what it passes in and `ossl_a2ulabel()` on the next line crashes if there is no terminator.
I could pass either to `ossl_punycode_decode()` since they are identical but I do need to alloc b and add the \0 for the next line.

Reply:
I prefer the call to ossl_punycode_decode be the not \0 terminated version.

Reply:
Done FWIW.